.github/workflows/release.yml

```diff
@@ -54,11 +60,18 @@ jobs:
         uses: rlespinasse/github-slug-action@v4
 
       - name: Set up Go
+        if: ${{ matrix.goos != 'windows' || matrix.os-variant != '7' }}
         uses: actions/setup-go@v5
         with:
           go-version: '^1.22.1'
 
-      - run: echo "basename=sni-${{env.GITHUB_REF_SLUG}}-${{matrix.goos}}-${{matrix.goarch}}${{matrix.alt}}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+      - name: Set up Go
+        if: ${{ matrix.goos == 'windows' && matrix.os-variant == '7' }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: '^1.20.14'
```

Comment:
```
Setup go version spec ^1.20.14
Found in cache @ C:\hostedtoolcache\windows\go\1.22.8\x64
Added go to the path
Successfully set up Go version ^1.20.14
C:\hostedtoolcache\windows\go\1.22.8\x64\bin\go.exe env GOMODCACHE
C:\hostedtoolcache\windows\go\1.22.8\x64\bin\go.exe env GOCACHE
C:\Users\runneradmin\go\pkg\mod
C:\Users\runneradmin\AppData\Local\go-build
Warning: Restore cache failed: Dependencies file is not found in D:\a\sni\sni. Supported file pattern: go.sum
go version go1.22.8 windows/amd64
```

Looks like the `^` is the lowest supported version and so it pulled the current version `1.22.8` when we want exactly `=1.20.14`.